cub/cub/util_type.cuh

```diff
@@ -73,6 +69,12 @@
 
 #include <cuda/std/type_traits>
 
+#if !defined(_LIBCUDACXX_COMPILER_NVRTC)
+#  include <iterator>
+#else
+#  include <cuda/std/iterator>
+#endif
```

Comment:
I am of the impression, that we are quite feature complete now with respect to iterator. what do we need from std?